bitcoin/src/blockdata/script/push_bytes.rs

```diff
@@ -293,23 +293,55 @@ impl PushBytes {
 
     /// Decodes an integer in script(minimal CScriptNum) format.
     ///
+    /// This code is based on the
+    /// [`CScriptNum` constructor in Bitcoin Core](https://github.com/bitcoin/bitcoin/blob/99a4ddf5ab1b3e514d08b90ad8565827fda7b63b/src/script/script.h#L245)
+    ///
+    /// # Errors
+    ///
+    /// * [`script::Error::NumericOverflow`] if result is not in range [-2^31 +1...2^31 -1].
+    /// * [`script::Error::NonMinimalPush`] if encoding is non-minimal.
```

Comment:
This is fine to go in like this but just for learnings sake this can also be written like this


```rust
    ///
    /// This code is based on the [`CScriptNum` constructor in Bitcoin Core].
    ///
    /// # Errors
    ///
    /// * [`script::Error::NumericOverflow`] if result is not in range [-2^39 +1...2^39 -1].
    /// * [`script::Error::NonMinimalPush`] if encoding is non-minimal.
    ///
    /// [`CScriptNum` constructor in Bitcoin Core]: <https://github.com/bitcoin/bitcoin/blob/99a4ddf5ab1b3e514d08b90ad8565827fda7b63b/src/script/script.h#L245>
```